test/type/explicit_existential.swift

```diff
@@ -156,6 +156,17 @@ func anyAny() {
 
 protocol P1 {}
 protocol P2 {}
+do {
+  // Test that we don't accidentally misparse an 'any' type as a 'some' type
+  // and vice versa.
+  let _: P1 & any P2 // expected-error {{'any' should appear at the beginning of a composition}}
+  let _: any P1 & any P2 // expected-error {{'any' should appear at the beginning of a composition}}
+  let _: any P1 & some P2 // expected-error {{'some' should appear at the beginning of a composition}}
```

Comment:
It probably warrants a tailored error message if someone uses `any` and `some` in the same protocol composition, but I'm not sure what it should say. I'll keep thinking about it

Reply:
I think the familiar "any/some type constrained to non-protocol, non-class type 'some/any T'" would already be a nice improvement.